src/firefly/java/edu/caltech/ipac/firefly/server/packagedata/ZipHandler.java

```diff
@@ -180,10 +181,16 @@ private static InputStream getInputStream(String filename, FileInfo fi, File bas
                 uc.setRequestProperty("Accept", "text/plain");
 
                 String extName = ofNullable(fi.getExternalName()).orElse("").trim();
+                String suffix = fi.getSuffix();
                 if (extName.isEmpty() || extName.endsWith("/")) {
                     String suggestedFilename = URLDownload.getSugestedFileName(uc);
                     suggestedFilename = isEmpty(suggestedFilename) ? getFileNameFromUrl(url) : suggestedFilename;
                     fi.setExternalName(extName + suggestedFilename);
+                    extName = extName + suggestedFilename;
```

Comment:
If you are going to do this, you should flip the lines around so that extName is not computed twice.
```.java
                    extName = extName + suggestedFilename;
                    fi.setExternalName(extName);
```

Reply:
You still need to handle duplicate files.

Reply:
@loitly about duplicates - since we're downloading files, any OS should be able to handle duplication, right? for example when I test with same file name on mac OS, it'll automatically add `-1`, `-2`, `-3` and so on at the end of the file name for duplicate file names. 

kind of like this: 

![Screenshot 2025-04-25 at 2 37 23 PM](https://github.com/user-attachments/assets/fc862c2a-4acd-4c14-8de7-4e2ae87a32d0)